x-pack/plugins/searchprofiler/public/application/hooks/use_indices.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useAppContext } from '../contexts/app_context';
+
+interface ReturnValue {
+  hasIndices: boolean;
+}
+
+export const useIndices = () => {
+  const { http } = useAppContext();
```

Comment:
Great, thank you! I is really useful!

Reply:
Another thing to think about is what should we do when if API call fails, should we surface anything to the users? what do you think? 🤔

Reply:
Sorry, I had skipped this comment. Well, I had set hasIndices to false if it failed, but if the user has indexes it is inconsistent. Do you know any example of something similar that we have in a plugin? How do we usually manage it?

Reply:
No worries! I don't think we've been very good at this sort of thing before, nor I can recall places in which we have similar scenarios to this. But perhaps is better to address this as a separate issue, perhaps involving design too as I'm not sure what the best way of visualizing this would be

Reply:
It make sense. It would be great to have some guidelines for this type of scenarios.